### PR TITLE
pipeline(ticdc):  sink node actor refactor

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve? 
this is a part of table pipeline actor model refactor #3881

### What is changed and how it works?
1. remove pipepline.NodeContext from some sink function definition
2. move replicaConfig to struct level

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: cdc/processor/pipeline/sink.go [hunk 1]
```diff
@@ -98,14 +103,20 @@ func (n *sinkNode) CheckpointTs() model.Ts { return atomic.LoadUint64(&n.checkpo
...
+	n.replicaConfig = replicaConfig
+	n.isTableActorMode = isTableActorMode
 	return nil
 }
 
 // stop is called when sink receives a stop command or checkpointTs reaches targetTs.
 // In this method, the builtin table sink will be closed by calling `Close`, and
 // no more events can be sent to this sink node afterwards.
-func (n *sinkNode) stop(ctx pipeline.NodeContext) (err error) {
+func (n *sinkNode) stop(ctx context.Context) (err error) {
```
Comment (asddongmen) [inline cdc/processor/pipeline/sink.go:119]: May be it is better to add a debug log here to indicate which table is closed?

**Comment by asddongmen** [comment]
rest LGTM

**Comment by ben1009** [comment]
any ut cover the change ?

File: cdc/processor/pipeline/sink.go (hunk 1 above)
Comment (sdojjy) [reply cdc/processor/pipeline/sink.go:119]: yes, I will add it in actor PR

**Review by asddongmen — APPROVED** [review]

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**